The certificate has expired when running `install_portaudio.sh` in Github Action

Hi team,

I am not sure if this is the correct place to post this issue. I am running a CI pipeline using Github actions which consists of buidling a docker image.

A portion of my Dockerfile
```
...

RUN mkdir -p /home/appuser/opt
WORKDIR /home/appuser/opt
#Commit on May 15, 2019 
ENV KALDI_SHA1 35f96db7082559a57dcc222218db3f0be6dd7983
RUN git clone https://github.com/kaldi-asr/kaldi && \
    cd /home/appuser/opt/kaldi && \
    git reset --hard $KALDI_SHA1 && \
    cd /home/appuser/opt/kaldi/tools && \
    make -j 2 && \
    ./install_portaudio.sh
```

This is part of my error log 
```
2022-06-06 21:17:42 (2.40 MB/s) - ‘sph2pipe_v2.5.tar.gz’ saved [329832/329832]
wget -T 10 -t 3 -O cub-1.8.0.zip https://github.com/NVlabs/cub/archive/1.8.0.zip
--2022-06-06 21:17:42--  https://github.com/NVlabs/cub/archive/1.8.0.zip
Resolving github.com (github.com)... 140.82.114.3
Connecting to github.com (github.com)|140.82.114.3|:443... connected.
302 Found
Location: https://us.openslr.org/resources/2/openfst-1.6.7.tar.gz [following]
--2022-06-06 21:17:42--  https://us.openslr.org/resources/2/openfst-1.6.7.tar.gz
Resolving us.openslr.org (us.openslr.org)... 46.101.158.64
Connecting to us.openslr.org (us.openslr.org)|46.101.158.64|:443... HTTP request sent, awaiting response... connected.
302 Found
Location: https://codeload.github.com/NVlabs/cub/zip/refs/tags/1.8.0 [following]
--2022-06-06 21:17:42--  https://codeload.github.com/NVlabs/cub/zip/refs/tags/1.8.0
Resolving codeload.github.com (codeload.github.com)... 140.82.113.10
Connecting to codeload.github.com (codeload.github.com)|140.82.113.10|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [application/zip]
Saving to: ‘cub-1.8.0.zip’
     0K .......... .......... .......... .......... .......... 6.76M
    50K .......... .......... .......... .......... .......... 5.19M
   100K .......... .......... .......... .......... .......... 5.88M
   150K .......... .......... .......... .......... .......... 6.18M
   200K .......... .......... .......... .......... .......... 6.32M
   250K .......... .......... .......... .......... .......... 7.28M
   300K .......... .......... .......... .......... .......... 6.65M
   350K .......... .......... .......... .......... ....ERROR: The certificate of ‘us.openslr.org’ is not trusted.
ERROR: The certificate of ‘us.openslr.org’ has expired.
..make: *** [Makefile:92: openfst-1.6.7.tar.gz] Error 5
.make: *** Waiting for unfinished jobs....
... 5.59M
   400K .................... .......... .......... .......... 5.89M
   450K .......... .......... .......... .......... .......... 5.67M
   500K .......... .......... .......... .......... .......... 6.02M
   550K .......... .......... .......... ........              7.81M=0.09s
2022-06-06 21:17:43 (6.17 MB/s) - ‘cub-1.8.0.zip’ saved [602396]
unzip -oq cub-1.8.0.zip
rm -f cub
ln -s cub-1.8.0 cub
The command '/bin/sh -c git clone https://github.com/kaldi-asr/kaldi &&     cd /home/appuser/opt/kaldi &&     git reset --hard $KALDI_SHA1 &&     cd /home/appuser/opt/kaldi/tools &&     make -j 2 &&     ./install_portaudio.sh' returned a non-zero code: 2
Error: Process completed with exit code 2.
```

The Dockerfile works fine in my local machine. I suspect that the job is taking too long, however, the job has only ran for 4minutes which is still below the limit for Github Action's allowed job duration. 

I wonder if anyone has any insights/solution for this ? Thanks !

I'll check. Probably some issue with letsencrypt.
y.

On Mon, Jun 6, 2022 at 4:35 PM Kai Kiat Poh ***@***.***>
wrote:

> Hi team,
>
> I am not sure if this is the correct place to post this issue. I am
> running a CI pipeline using Github actions which consists of buidling a
> docker image.
>
> A portion of my Dockerfile
>
> ...
>
>
>
> RUN mkdir -p /home/appuser/opt
>
> WORKDIR /home/appuser/opt
>
> #Commit on May 15, 2019
>
> ENV KALDI_SHA1 35f96db7082559a57dcc222218db3f0be6dd7983
>
> RUN git clone https://github.com/kaldi-asr/kaldi && \
>
>     cd /home/appuser/opt/kaldi && \
>
>     git reset --hard $KALDI_SHA1 && \
>
>     cd /home/appuser/opt/kaldi/tools && \
>
>     make -j 2 && \
>
>     ./install_portaudio.sh
>
>
> This is part of my error log
>
> 2022-06-06 21:17:42 (2.40 MB/s) - ‘sph2pipe_v2.5.tar.gz’ saved [329832/329832]
>
> wget -T 10 -t 3 -O cub-1.8.0.zip https://github.com/NVlabs/cub/archive/1.8.0.zip
>
> --2022-06-06 21:17:42--  https://github.com/NVlabs/cub/archive/1.8.0.zip
>
> Resolving github.com (github.com)... 140.82.114.3
>
> Connecting to github.com (github.com)|140.82.114.3|:443... connected.
>
> 302 Found
>
> Location: https://us.openslr.org/resources/2/openfst-1.6.7.tar.gz [following]
>
> --2022-06-06 21:17:42--  https://us.openslr.org/resources/2/openfst-1.6.7.tar.gz
>
> Resolving us.openslr.org (us.openslr.org)... 46.101.158.64
>
> Connecting to us.openslr.org (us.openslr.org)|46.101.158.64|:443... HTTP request sent, awaiting response... connected.
>
> 302 Found
>
> Location: https://codeload.github.com/NVlabs/cub/zip/refs/tags/1.8.0 [following]
>
> --2022-06-06 21:17:42--  https://codeload.github.com/NVlabs/cub/zip/refs/tags/1.8.0
>
> Resolving codeload.github.com (codeload.github.com)... 140.82.113.10
>
> Connecting to codeload.github.com (codeload.github.com)|140.82.113.10|:443... connected.
>
> HTTP request sent, awaiting response... 200 OK
>
> Length: unspecified [application/zip]
>
> Saving to: ‘cub-1.8.0.zip’
>
>      0K .......... .......... .......... .......... .......... 6.76M
>
>     50K .......... .......... .......... .......... .......... 5.19M
>
>    100K .......... .......... .......... .......... .......... 5.88M
>
>    150K .......... .......... .......... .......... .......... 6.18M
>
>    200K .......... .......... .......... .......... .......... 6.32M
>
>    250K .......... .......... .......... .......... .......... 7.28M
>
>    300K .......... .......... .......... .......... .......... 6.65M
>
>    350K .......... .......... .......... .......... ....ERROR: The certificate of ‘us.openslr.org’ is not trusted.
>
> ERROR: The certificate of ‘us.openslr.org’ has expired.
>
> ..make: *** [Makefile:92: openfst-1.6.7.tar.gz] Error 5
>
> .make: *** Waiting for unfinished jobs....
>
> ... 5.59M
>
>    400K .................... .......... .......... .......... 5.89M
>
>    450K .......... .......... .......... .......... .......... 5.67M
>
>    500K .......... .......... .......... .......... .......... 6.02M
>
>    550K .......... .......... .......... ........              7.81M=0.09s
>
> 2022-06-06 21:17:43 (6.17 MB/s) - ‘cub-1.8.0.zip’ saved [602396]
>
> unzip -oq cub-1.8.0.zip
>
> rm -f cub
>
> ln -s cub-1.8.0 cub
>
> The command '/bin/sh -c git clone https://github.com/kaldi-asr/kaldi &&     cd /home/appuser/opt/kaldi &&     git reset --hard $KALDI_SHA1 &&     cd /home/appuser/opt/kaldi/tools &&     make -j 2 &&     ./install_portaudio.sh' returned a non-zero code: 2
>
> Error: Process completed with exit code 2.
>
>
> The Dockerfile works fine in my local machine. I suspect that the job is
> taking too long, however, the job has only ran for 4minutes which is still
> below the limit for Github Action's allowed job duration.
>
> I wonder if anyone has any insights/solution for this ? Thanks !
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/issues/4755>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACUKYXY5EN7HQ2B24D47NWTVNYECHANCNFSM5X7SJYRQ>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

IIRC some older versions of wget had an issue after LE did some magic with
replacing the root authority?
perhaps make sure you have reasonably new OS in the docker/new wget? On
what OS you build the docker image?
y.

On Mon, Jun 6, 2022 at 5:17 PM Jan Yenda Trmal ***@***.***> wrote:

> Seems it's ok on our side:
> ```
>   Certificate Name: www.openslr.org
>     Key Type: RSA
>     Domains: www.openslr.org openslr.org us.openslr.org
>     Expiry Date: 2022-08-23 15:01:56+00:00 (VALID: 77 days)
> ```
>
> On Mon, Jun 6, 2022 at 5:03 PM Jan Yenda Trmal ***@***.***> wrote:
>
>> I'll check. Probably some issue with letsencrypt.
>> y.
>>
>> On Mon, Jun 6, 2022 at 4:35 PM Kai Kiat Poh ***@***.***>
>> wrote:
>>
>>> Hi team,
>>>
>>> I am not sure if this is the correct place to post this issue. I am
>>> running a CI pipeline using Github actions which consists of buidling a
>>> docker image.
>>>
>>> A portion of my Dockerfile
>>>
>>> ...
>>>
>>>
>>>
>>> RUN mkdir -p /home/appuser/opt
>>>
>>> WORKDIR /home/appuser/opt
>>>
>>> #Commit on May 15, 2019
>>>
>>> ENV KALDI_SHA1 35f96db7082559a57dcc222218db3f0be6dd7983
>>>
>>> RUN git clone https://github.com/kaldi-asr/kaldi && \
>>>
>>>     cd /home/appuser/opt/kaldi && \
>>>
>>>     git reset --hard $KALDI_SHA1 && \
>>>
>>>     cd /home/appuser/opt/kaldi/tools && \
>>>
>>>     make -j 2 && \
>>>
>>>     ./install_portaudio.sh
>>>
>>>
>>> This is part of my error log
>>>
>>> 2022-06-06 21:17:42 (2.40 MB/s) - ‘sph2pipe_v2.5.tar.gz’ saved [329832/329832]
>>>
>>> wget -T 10 -t 3 -O cub-1.8.0.zip https://github.com/NVlabs/cub/archive/1.8.0.zip
>>>
>>> --2022-06-06 21:17:42--  https://github.com/NVlabs/cub/archive/1.8.0.zip
>>>
>>> Resolving github.com (github.com)... 140.82.114.3
>>>
>>> Connecting to github.com (github.com)|140.82.114.3|:443... connected.
>>>
>>> 302 Found
>>>
>>> Location: https://us.openslr.org/resources/2/openfst-1.6.7.tar.gz [following]
>>>
>>> --2022-06-06 21:17:42--  https://us.openslr.org/resources/2/openfst-1.6.7.tar.gz
>>>
>>> Resolving us.openslr.org (us.openslr.org)... 46.101.158.64
>>>
>>> Connecting to us.openslr.org (us.openslr.org)|46.101.158.64|:443... HTTP request sent, awaiting response... connected.
>>>
>>> 302 Found
>>>
>>> Location: https://codeload.github.com/NVlabs/cub/zip/refs/tags/1.8.0 [following]
>>>
>>> --2022-06-06 21:17:42--  https://codeload.github.com/NVlabs/cub/zip/refs/tags/1.8.0
>>>
>>> Resolving codeload.github.com (codeload.github.com)... 140.82.113.10
>>>
>>> Connecting to codeload.github.com (codeload.github.com)|140.82.113.10|:443... connected.
>>>
>>> HTTP request sent, awaiting response... 200 OK
>>>
>>> Length: unspecified [application/zip]
>>>
>>> Saving to: ‘cub-1.8.0.zip’
>>>
>>>      0K .......... .......... .......... .......... .......... 6.76M
>>>
>>>     50K .......... .......... .......... .......... .......... 5.19M
>>>
>>>    100K .......... .......... .......... .......... .......... 5.88M
>>>
>>>    150K .......... .......... .......... .......... .......... 6.18M
>>>
>>>    200K .......... .......... .......... .......... .......... 6.32M
>>>
>>>    250K .......... .......... .......... .......... .......... 7.28M
>>>
>>>    300K .......... .......... .......... .......... .......... 6.65M
>>>
>>>    350K .......... .......... .......... .......... ....ERROR: The certificate of ‘us.openslr.org’ is not trusted.
>>>
>>> ERROR: The certificate of ‘us.openslr.org’ has expired.
>>>
>>> ..make: *** [Makefile:92: openfst-1.6.7.tar.gz] Error 5
>>>
>>> .make: *** Waiting for unfinished jobs....
>>>
>>> ... 5.59M
>>>
>>>    400K .................... .......... .......... .......... 5.89M
>>>
>>>    450K .......... .......... .......... .......... .......... 5.67M
>>>
>>>    500K .......... .......... .......... .......... .......... 6.02M
>>>
>>>    550K .......... .......... .......... ........              7.81M=0.09s
>>>
>>> 2022-06-06 21:17:43 (6.17 MB/s) - ‘cub-1.8.0.zip’ saved [602396]
>>>
>>> unzip -oq cub-1.8.0.zip
>>>
>>> rm -f cub
>>>
>>> ln -s cub-1.8.0 cub
>>>
>>> The command '/bin/sh -c git clone https://github.com/kaldi-asr/kaldi &&     cd /home/appuser/opt/kaldi &&     git reset --hard $KALDI_SHA1 &&     cd /home/appuser/opt/kaldi/tools &&     make -j 2 &&     ./install_portaudio.sh' returned a non-zero code: 2
>>>
>>> Error: Process completed with exit code 2.
>>>
>>>
>>> The Dockerfile works fine in my local machine. I suspect that the job is
>>> taking too long, however, the job has only ran for 4minutes which is still
>>> below the limit for Github Action's allowed job duration.
>>>
>>> I wonder if anyone has any insights/solution for this ? Thanks !
>>>
>>> —
>>> Reply to this email directly, view it on GitHub
>>> <https://github.com/kaldi-asr/kaldi/issues/4755>, or unsubscribe
>>> <https://github.com/notifications/unsubscribe-auth/ACUKYXY5EN7HQ2B24D47NWTVNYECHANCNFSM5X7SJYRQ>
>>> .
>>> You are receiving this because you are subscribed to this thread.Message
>>> ID: ***@***.***>
>>>
>>

Hi team,

I changed the os version to the debian:latest, but I am getting this message
```
Warning: IRSTLM is not installed by default anymore. If you need IRSTLM
Warning: use the script extras/install_irstlm.sh
All done OK.
****() Installing portaudio
Could not find portaudio tarball pa_stable_v19_20111121.tgz
Trying to download it via wget!
--2022-06-07 00:41:32--  http://www.portaudio.com/archives/pa_stable_v19_20111121.tgz
Resolving www.portaudio.com (www.portaudio.com)... 107.22.210.58
Connecting to www.portaudio.com (www.portaudio.com)|107.22.210.58|:80... failed: Connection timed out.
Retrying.
--2022-06-07 00:41:43--  (try: 2)  http://www.portaudio.com/archives/pa_stable_v19_20111121.tgz
Connecting to www.portaudio.com (www.portaudio.com)|107.22.210.58|:80... failed: Connection timed out.
Retrying.
--2022-06-07 00:41:55--  (try: 3)  http://www.portaudio.com/archives/pa_stable_v19_20111121.tgz
Connecting to www.portaudio.com (www.portaudio.com)|107.22.210.58|:80... failed: Connection timed out.
Giving up.
Download of pa_stable_v19_20111121.tgz - failed!
Aborting script. Please download and install port audio manually!
The command '/bin/sh -c git clone https://github.com/kaldi-asr/kaldi &&     cd /home/appuser/opt/kaldi &&     git reset --hard $KALDI_SHA1 &&     cd /home/appuser/opt/kaldi/tools &&     make -j 2 &&     ./install_portaudio.sh' returned a non-zero code: 1
Error: Process completed with exit code 1.

```

How do I install the port audio manually ?

--- a/tools/extras/install_portaudio.sh
+++ b/tools/extras/install_portaudio.sh
@@ -43,31 +43,32 @@ portaudio_tarball="pa_stable_${VERSION}.tgz"
 portaudio_github_tarball="pa_stable_${VERSION}_r1788.tar.gz"
 
 if [ ! -e $portaudio_tarball ]; then
-    echo "Could not find portaudio tarball $portaudio_tarball locally, downloading it..."
+  echo "Could not find portaudio tarball $portaudio_tarball locally, downloading it..."
 
-    if [ -d "$DOWNLOAD_DIR" ]; then
-        cp -p "$DOWNLOAD_DIR/$portaudio_tarball" .
-    else
-        if ! $WGET --version >&/dev/null; then
-            echo "This script requires you to first install wget"
-            echo "You can also just download pa_stable_$VERSION.tgz from"
-            echo "http://www.portaudio.com/download.html)"
-            exit 1;
-        fi
-	$WGET -nv -T 10 -t 3 -O $portaudio_tarball https://github.com/PortAudio/portaudio/archive/refs/tags/${portaudio_github_tarball} || \
-	$WGET -nv -T 10 -t 3 -O $portaudio_tarball http://www.portaudio.com/archives/$portaudio_tarball || \
-	rm ${portaudio_tarball}
-
+  if [ -d "$DOWNLOAD_DIR" ]; then
+    cp -p "$DOWNLOAD_DIR/$portaudio_tarball" .
+  else
+    if ! $WGET --version >&/dev/null; then
+      echo "This script requires you to first install wget"
+      echo "You can also just download pa_stable_$VERSION.tgz from"
+      echo "http://www.portaudio.com/download.html)"
+      exit 1;
     fi
 
-    if [ ! -e $portaudio_tarball ]; then
-        echo "Download of $portaudio_tarball - failed."
-        echo "Aborting script. Please download and install port audio manually."
-        exit 1;
-    fi
+    $WGET --no-check-certificate -nv -T 10 -t 3 -O $portaudio_tarball https://github.com/PortAudio/portaudio/archive/refs/tags/${portaudio_github_tarball} || \
+      $WGET --no-check-certificate -nv -T 10 -t 3 -O $portaudio_tarball http://files.portaudio.com/archives/$portaudio_tarball || \
+      rm ${portaudio_tarball}
+
+  fi
+
+  if [ ! -e $portaudio_tarball ]; then
+    echo "Download of $portaudio_tarball - failed."
+    echo "Aborting script. Please download and install port audio manually."
+    exit 1;
+  fi
 fi
 
-mkdir portaudio && tar -xzf $portaudio_tarball -C portaudio --strip-components 1 || exit 1
+mkdir -p  portaudio && tar -xzf $portaudio_tarball -C portaudio --strip-components 1 || exit 1
 
 read -d '' pa_patch << "EOF"
 --- portaudio/Makefile.in	2012-08-05 10:42:05.000000000 +0300

